vercel/speed-insights
# feat: Astro support

### 📓 What's in there?

This PR brings Astro first-class support with a dedicated component.

### 🧪 How to test?

There's a new example application.
1. build your package locally `pnpm -F @vercel/speed-insights build`
2. get new deps `pnpm i`
3. edit the example app, to see it sending metrics to the production intake (will 404 because of the lack or route, but you'll see the payloads): edit `apps/astro/src/components/BaseHead.astro` to add `<SpeedInsights scriptSrc="https://va.vercel-scripts.com/v1/speed-insights/script.js">`
4. run the app: `cd apps/astro && pnpm dev`
5. visit localhost:4321 with your dev tools open and see the reports with correct framework name and route.

### ❗ Notes to reviewers

This component is one-of-its-kind: we don't want to bundle it with `tsup` like for Vue, Sveltekit, React or Next.
Astro will care of packaging it when it'll bundle the customer's astro app.

So this index.astro file is copied after running tsup directly into `dist/`

**This is draft until we find a way to pass  `beforeSend` to `injectSpeedInsights()` utility**
